corehq/apps/locations/permissions.py

```diff
@@ -319,6 +319,17 @@ def user_can_access_any_location_id(domain, user, location_ids):
             .exists())
 
 
+def user_can_access_other_user(domain, user, other_user):
+    if user.has_permission(domain, 'access_all_locations'):
+        return True
+
+    accessible_location_ids = (SQLLocation.objects
+        .accessible_to_user(domain, user)
+        .values_list('id', flat=True))
+
+    return other_user.sql_locations.filter(id__in=accessible_location_ids).count() > 0
```

Comment:
> what is the status of webusers and locations? should that be first class support?

Yup.  The distinction between web and mobile users is pretty blurry, I'd support @proteusvacuum 's idea of dropping the `sql_locations` method and striving for API compatability.  Web users can edit and archive form submissions, and mobile workers can access some pages on HQ.  Additionally, we usually use `CouchUser.get` methods, which can return either, so it's hard to tell which one you may get.